applications/callflow/src/cf_endpoint.erl

```diff
@@ -1197,9 +1197,26 @@ generate_sip_headers(Endpoint, Acc, Call) ->
     HeaderFuns = [fun maybe_add_sip_headers/1
                   ,fun(J) -> maybe_add_alert_info(J, Endpoint, Inception) end
                   ,fun(J) -> maybe_add_aor(J, Endpoint, Call) end
+                  ,fun(J) -> maybe_add_diversion(J, Call, Inception, Endpoint) end
                  ],
     lists:foldr(fun(F, JObj) -> F(JObj) end, Acc, HeaderFuns).
 
+-spec maybe_add_diversion(wh_json:object(), whapps_call:call(), api_binary(), wh_json:object()) -> wh_json:object().
+maybe_add_diversion(JObj, _Call, 'undefined', _Endpoint) ->
+    JObj;
+maybe_add_diversion(JObj, Call, Inception, Endpoint) ->
+    ShouldAddDiversion = whapps_call:authorizing_id(Call) =:= 'undefined'
```

Comment:
Can we add a check to `whapps_config:get_is_true(?CONFIG_CAT, <<"should_add_call_forward_diversion">>, 'false')` to preserve existing functionality?